File: js/busqueda_binaria_pra.js
function busquedaBinaria(lista, objetivo) {
    let izquierda = 0;
    let derecha = lista.length - 1;

    while (izquierda <= derecha) {
        let medio = Math.floor((izquierda + derecha) / 2);

        if (lista[medio] === objetivo) {
            return medio;
        } else if (lista[medio] < objetivo) {
            izquierda = medio + 1;
        } else {
            derecha = medio - 1;
        }
    }

    return -1;
}

let numeros = [1, 3, 5, 7, 9];

console.log(busquedaBinaria(numeros, 7,));

File: python/busqueda_binaria_pra.py
def busqueda_binaria(lista, objetivo):
    izquierda = 0
    derecha = len(lista) - 1

    while izquierda <= derecha:
        medio = (izquierda + derecha) // 2

        if lista[medio] == objetivo:
            return medio

        elif lista[medio] < objetivo:
            izquierda = medio + 1

        else:
            derecha = medio - 1

    return -1


numeros = [1, 3, 5, 7, 9]

print(busqueda_binaria(numeros, 7))